plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -189,22 +191,37 @@ public function update( &$order ) {
 
 		// Also grab the current status so we can compare.
 		$previous_status = get_post_status( $order->get_id() );
+		// If the order doesn't exist in the DB, we will consider it as new.
+		if ( ! $previous_status ) {
```

Comment:
get_post_status can be false when there's a intermittent DB issue, and not necessarily if the post does not exist. A better check for not existing in DB is to check fo `$order->get_id() > 0`

Reply:
could this be `( ! $previous_status && $order->get_id() === 0 )` or is it too redundant? Since we have the bool already, it would be quick to evaluate and save the `get_id()` call.
We're only changing order status in a non-persistent way here to allow the triggering of the hook, as per Mike's comment

Reply:
I added the check on https://github.com/woocommerce/woocommerce/pull/49098/commits/1ab534aabc39613aa8f21e872a0b3087b2b2c0d0 in case you are onboard with it and decide to approve 👍🏼